include/oneapi/dpl/pstl/hetero/numeric_impl_hetero.h

```diff
@@ -120,7 +120,9 @@ __pattern_transform_scan_base_impl(_ExecutionPolicy&& __exec, _Iterator1 __first
     if ((__n_uniform & (__n_uniform - 1)) != 0)
         __n_uniform = __par_backend_hetero::__dpl_bit_floor(__n) << 1;
 
-    const auto __max_slm_size = __exec.queue().get_device().template get_info<sycl::info::device::local_mem_size>();
+    // Pessimistically only use half of the memory to take into account memory used by compiled kernel
+    const auto __max_slm_size =
+        __exec.queue().get_device().template get_info<sycl::info::device::local_mem_size>() * 0.5;
```

Comment:
Similarly to sort, should this be explicitly `::std::size_t` instead of auto?

Reply:
Yes, just changed.